--- src/diffDOM/virtual/apply.js	1985-10-26 09:15:00.000000000 +0100
+++ src/diffDOM/virtual/apply.js	2020-05-04 15:41:48.137362273 +0200
@@ -26,7 +26,7 @@
 function applyVirtualDiff(
         tree,
         diff,
-        options // {preDiffApply, postDiffApply, _const}
+        options // {preVirtualDiffApply, postVirtualDiffApply, _const}
     ) {
     const routeInfo = getFromVirtualRoute(tree, diff[options._const.route])
     let node = routeInfo.node
@@ -40,7 +40,7 @@
         node
     }
 
-    if (options.preDiffApply(info)) {
+    if (options.preVirtualDiffApply(info)) {
         return true
     }
 
@@ -277,7 +277,7 @@
 
     // capture newNode for the callback
     info.newNode = newNode
-    options.postDiffApply(info)
+    options.postVirtualDiffApply(info)
 
     return
 }
