--- index.js.orig	2023-02-02 11:31:05.190912936 +0100
+++ index.js	2023-02-02 13:45:33.245247337 +0100
@@ -2539,7 +2539,8 @@
             let rule = this.tags[i];
             if (matches(dom, rule.tag) &&
                 (rule.namespace === undefined || dom.namespaceURI == rule.namespace) &&
-                (!rule.context || context.matchesContext(rule.context))) {
+                (!rule.context || context.matchesContext(rule.context)) &&
+                context.matchesType(this.schema, rule)) {
                 if (rule.getAttrs) {
                     let result = rule.getAttrs(dom);
                     if (result === false)
@@ -3104,6 +3105,45 @@
         };
         return match(parts.length - 1, this.open);
     }
+    matchesType(schema, rule) {
+        let type;
+        if (rule.node) {
+            type = schema.nodes[rule.node];
+        } else if (rule.mark) {
+            type = schema.marks[rule.mark];
+            return true;
+        } else {
+            return true;
+        }
+        const option = this.options.context;
+        const useRoot = !this.isOpen && (!option || option.parent.type == this.nodes[0].type);
+        const minDepth = -(option ? option.depth + 1 : 0) + (useRoot ? 0 : 1);
+        const nextMatch = function(contentMatch) {
+            if (contentMatch.matchType(type)) return true;
+            if (!contentMatch.validEnd) {
+                for (const next of contentMatch.next) {
+                    const hasMatch = nextMatch(next.next);
+                    if (hasMatch) return true;
+                }
+            }
+            return false;
+        };
+        const match = (depth) => {
+            let next = depth > 0 || (depth == 0 && useRoot) ? this.nodes[depth]
+                : option && depth >= minDepth ? option.node(depth - minDepth)
+                : null;
+            if (!next) return false;
+            if (!next.match) {
+                next = next.type.contentMatch;
+            } else {
+                next = next.match;
+            }
+            const hasMatch = nextMatch(next);
+            if (hasMatch) return hasMatch;
+            return match(depth - 1);
+        };
+        return match(this.open);
+    }
     textblockFromContext() {
         let $context = this.options.context;
         if ($context)
