--- index.es.js	1985-10-26 09:15:00.000000000 +0100
+++ index.es.js	2021-09-08 15:19:31.289320573 +0200
@@ -3348,6 +3351,7 @@
 function checkCSS(view) {
   if (cssChecked) { return }
   cssChecked = true;
+  if (view.cssChecked) { return }
   if (getComputedStyle(view.dom).whiteSpace == "normal")
     { console["warn"]("ProseMirror expects the CSS white-space property to be set, preferably to 'pre-wrap'. It is recommended to load style/prosemirror.css from the prosemirror-view package."); }
 }
