--- dist/index.js	1985-10-26 09:15:00.000000000 +0100
+++ dist/index.j	2020-02-18 18:14:21.830142338 +0100
@@ -3096,7 +3096,7 @@
 var cssChecked = false;
 
 function checkCSS(view) {
-  if (cssChecked) { return }
+  if (cssChecked || view.cssChecked) { return }
   cssChecked = true;
   if (getComputedStyle(view.dom).whiteSpace == "normal")
     { console["warn"]("ProseMirror expects the CSS white-space property to be set, preferably to 'pre-wrap'. It is recommended to load style/prosemirror.css from the prosemirror-view package."); }
