--- index.es.js     2022-02-15 22:15:32.717742784 +0100
+++ index.es.js 2022-02-15 22:15:49.869371930 +0100
@@ -64,7 +64,7 @@
     if (node == targetNode && off == targetOff) { return true }
     if (off == (dir < 0 ? 0 : nodeSize(node))) {
       var parent = node.parentNode;
-      if (parent.nodeType != 1 || hasBlockDesc(node) || atomElements.test(node.nodeName) || node.contentEditable == "false")
+      if (!parent || parent.nodeType != 1 || hasBlockDesc(node) || atomElements.test(node.nodeName) || node.contentEditable == "false")
         { return false }
       off = domIndex(node) + (dir < 0 ? 0 : 1);
       node = parent;
