--- dist/dropcursor.js	1985-10-26 09:15:00.000000000 +0100
+++ dist/dropcursor.js.new	2019-11-17 22:34:56.333009128 +0100
@@ -89,7 +89,7 @@
     rect = {left: coords.left - this.width / 2, right: coords.left + this.width / 2, top: coords.top, bottom: coords.bottom};
   }
 
-  var parent = this.editorView.dom.offsetParent;
+  var parent = this.editorView.dom.offsetParent || this.editorView.dom.parentNode;
   if (!this.element) {
     this.element = parent.appendChild(document.createElement("div"));
     if (this.class) { this.element.className = this.class; }
